code/game/objects/items/food/pastries.dm

```diff
@@ -585,12 +585,26 @@
 	name = "fortune cookie"
 	desc = "A true prophecy in each cookie!"
 	icon_state = "fortune_cookie"
+	trash_type = /obj/item/paper
+	///This is the fortune in the cookie
+	var/obj/item/paper/fortune
 	food_reagents = list(/datum/reagent/consumable/nutriment = 5)
 	tastes = list("cookie" = 1)
 	foodtypes = GRAIN | SUGAR
 	food_flags = FOOD_FINGER_FOOD
 	w_class = WEIGHT_CLASS_SMALL
 
+/obj/item/food/fortunecookie/proc/get_fortune()
+	// If a fortune exists, use that.
+	if (fortune)
+		return fortune
+	// Otherwise, make a blank page.
+	var/obj/item/paper/out_paper = new ()
+	return out_paper
```

Comment:
You can use locate() to find items stored in some other object's contents. In this case, something like `var/obj/item/paper/fortune = locate(/obj/item/paper) in src` would get you what you want.
See the ref https://secure.byond.com/docs/ref/#/proc/locate

Oh and regardless of all that, make sure you move your created paper to `drop_location()`, since we're not guaranteed to be making trash in someone's hands (This is more a food element issue, but it is an actual one)
![image](https://user-images.githubusercontent.com/58055496/148858779-783ff19b-991a-46ab-a0ec-e9eebc760302.png)

Reply:
had no idea about locate()! thanks.